--- openssl.original/Makefile.shared	2018-03-28 00:50:37.000000000 +1100
+++ openssl/Makefile.shared	2018-04-12 10:56:44.000000000 +1000
@@ -228,7 +228,7 @@
 link_shlib.darwin:
 	@ ALLSYMSFLAGS='-all_load'; \
 	NOALLSYMSFLAGS=''; \
-	SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -current_version $(SHLIBVERSION) -compatibility_version $(SHLIBVERSION) -install_name $(INSTALLTOP)/$(LIBDIR)/$(SHLIBNAME_FULL)"; \
+   SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -current_version $(SHLIBVERSION) -compatibility_version $(SHLIBVERSION) -install_name @rpath/$(SHLIBNAME_FULL) -Wl,-rpath,@executable_path/ -Wl,-rpath,@loader_path/"; \
 	$(LINK_SO_SHLIB)
 link_app.darwin:	# is there run-path on darwin?
 	$(LINK_APP)
